state/application.go

```diff
@@ -1949,64 +1949,46 @@ func (a *Application) SetDownloadedIDAndHash(id, hash string) error {
 	return nil
 }
 
-func checkSeriesForSetCharm(currentPlatform *Platform, charm *Charm, ForceBase bool) error {
+func checkSeriesForSetCharm(currentPlatform *Platform, ch *Charm, ForceBase bool) error {
 	// For old style charms written for only one series, we still retain
 	// this check. Newer charms written for multi-series have a URL
 	// with series = "".
-	curSeries, err := corebase.GetSeriesFromChannel(currentPlatform.OS, currentPlatform.Channel)
+	curBase, err := corebase.ParseBase(currentPlatform.OS, currentPlatform.Channel)
 	if err != nil {
 		return errors.Trace(err)
 	}
-	charmSeries, err := corecharm.ComputedSeries(charm)
+	charmBases, err := corecharm.ComputedBases(ch)
 	if err != nil {
 		return errors.Trace(err)
 	}
-	if charm.URL().Series != "" {
-		// Allow series change when switching to charmhub charms.
-		// Account for legacy charms with "kubernetes" series in the URL.
-		if charm.URL().Schema != "ch" && charm.URL().Series != corebase.Kubernetes.String() && charm.URL().Series != curSeries {
-			return errors.Errorf("cannot change an application's series")
-		}
-	} else if !ForceBase {
+	if len(charmBases) == 0 {
+		return errors.NewNotValid(nil, fmt.Sprintf("charm %q does not support any bases. Not valid", ch.Meta().Name))
```

Comment:
```suggestion
		return errors.NotValidf("charm %q does not support any bases", ch.Meta().Name)
```

Reply:
I think this is in a few places. I'll change throughout